src/cryptojwt/key_bundle.py

```diff
@@ -251,6 +266,12 @@ def _do_local(self, kid):
 
     def _local_update_required(self) -> bool:
         stat = os.stat(self.source)
+        print("@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@")
+        print("mtime:", stat.st_mtime)
+        print("last_local:", self.last_local)
+        if self.last_local:
+            print("mtime < last_local", stat.st_mtime < self.last_local)
+        print("@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@")
         if self.last_local and stat.st_mtime < self.last_local:
```

Comment:
re: https://github.com/IdentityPython/JWTConnect-Python-CryptoJWT/pull/66#issuecomment-689498564

@janste63 it is a (weird) timing issue - this is where the problem occurs: `stat.st_mtime < self.last_local` is `True` and thus the file is not read and the keys are not loaded.

see the build matrix - some tests pass, some fail, restarting a build may change the result..
https://travis-ci.org/github/IdentityPython/JWTConnect-Python-CryptoJWT/builds/726033878

let's focus on: https://travis-ci.org/github/IdentityPython/JWTConnect-Python-CryptoJWT/jobs/726033881#L1070-L1117

following what is happening:
- `tests/test_03_key_bundle.py::test_update_2` is starting, it creates a key and writes it to a file
- `kb = KeyBundle(source="file://{}".format(fname), fileformat="jwks")` then the KeyBundle object is initialized, which during initialization will load the keys from the file. 
- `do_local_jwk` is called, which invokes `_local_update_required` which sets `last_local` to the file's `mtime`.
- `do_local_jwk` will then continue to load the keys and then set `last_local` to `time.time()`

back to the `test_update_2`
- another key is created and the two keys are written to the same file
- the `KeyBundle` object is called to `update()` its state
- `do_local_jwk` is called, which invokes `_local_update_required` which reads the file's `mtime`.
- !! At this point is where we see the weird behaviour: the `mtime` is _before_ the `last_local` even though the file was changed _after_ `last_local` was set !!

Something seems wrong with the filesystem properties, and I am guessing this might be because of TravisCI optimizations on their service - maybe, I can't be sure. Do check it out yourself, too, and let me know if see something different.

I'd say that looking at `mtime` to figure out if we should read the file is probably not the best approach. Do we have a better option? I am not sure, atm.
This PR introduces the calls to `_local_update_required` to check this. Maybe, we can skip those checks and let the `do_local_xxx` methods always read the file.

Reply:
I now pushed another approach that gets rid the dependency on mtime. Instead, what the KeyBundle does is, it calculates a checksum (currently based on sha256). This makes sure that we process the file only when it actually changed, but I'm not sure if it's any more efficient than just doing the processing directly (no numbers, but reading the file contents and calculating a hash is not cheap).

Reply:
I think I disagree with this solution. Relying on mtime is perfectly good, and we should fix the tests on Travis if they fail. Rereading the entire file every time is very inefficient compared to a simple stat. File system sematics are very clear.

Reply:
I understand your position - I will revert this. However, the facts around the mtime problem are described above. If you have an idea to fix it let me know.

Reply:
`time.sleep(1)` ? :-)

Reply:
Perfectly reasonable for a test case.

Reply:
Reverted, cleaned up, rebased and now green ;)